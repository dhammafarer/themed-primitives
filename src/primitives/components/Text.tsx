import * as React from "react";
import { css, styled } from "../../theme";
import { Box, box, BoxProps, ResponsiveScale, ResponsiveString } from "./Box";
import {
  getFontSize,
  getFontWeight,
  getFontFamily,
  getLineHeight,
  getLetterSpacing,
  getTextTransform,
  getTextAlign,
} from "../getters";

const text = css<TextProps>`
  font-family: ${props => props.theme.fonts.sans};
  ${box}
  ${props => css`
    ${getFontSize(props)}
    ${getFontWeight(props)}
    ${getFontFamily(props)}
    ${getLineHeight(props)}
    ${getLetterSpacing(props)}
    ${getTextTransform(props)}
    ${getTextAlign(props)}
  `}
`;

interface TextProps extends BoxProps {
  fontFamily?: string;
  fontSize?: ResponsiveScale;
  fontWeight?: ResponsiveScale;
  shadow?: ResponsiveScale;
  textTransform?: ResponsiveString;
  textAlign?: ResponsiveString;
  lineHeight?: ResponsiveString;
  letterSpacing?: ResponsiveString;
}

const Text = styled(
  ({
    fontFamily,
    fontSize,
    fontWeight,
    shadow,
    textTransform,
    textAlign,
    lineHeight,
    letterSpacing,
    ...rest
  }) => <Box {...rest} />
)<TextProps & React.HTMLProps<HTMLDivElement>>`
  ${text}
`;

export { Text, TextProps };
