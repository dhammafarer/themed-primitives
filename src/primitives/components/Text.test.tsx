import * as React from "react";
import * as renderer from "react-test-renderer";

import { Text } from "./Text";

it("should render a div", () => {
  const tree = renderer.create(<Text />).toJSON();

  expect(tree).toMatchSnapshot();
});
