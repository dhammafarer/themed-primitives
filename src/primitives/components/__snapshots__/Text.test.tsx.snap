// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a div 1`] = `
<div
  className="sc-bwzfXH iURqys sc-bdVaJa jolpEI"
/>
`;
